examples/smac/feedforward/decentralised/calculate_absolute_metric.py

```diff
@@ -0,0 +1,128 @@
+from absl import app, flags
+
+from mava.systems import ippo
+from mava.utils.environments.smac_utils import make_environment
+from mava.utils.loggers import logger_utils
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string(
+    "map_name",
+    "2m_vs_1z",
+    "Starcraft 2 micromanagement map name (str).",
+)
+
+flags.DEFINE_string(
+    "mava_id",
+    str(datetime.now()),
+    "Experiment identifier that can be used to continue experiments.",
+)
+flags.DEFINE_string("base_dir", "~/mava", "Base dir to store experiments.")
+
+
+def main(_: Any) -> None:
+    """Example running feedforward IPPO on SMAC environment."""
+
+    # Environment
+    environment_factory = functools.partial(make_environment, map_name=FLAGS.map_name)
+
+    # Networks.
+    def network_factory(*args: Any, **kwargs: Any) -> Any:
+        return ippo.make_default_networks(  # type: ignore
+            policy_layer_sizes=(64, 64),
+            critic_layer_sizes=(64, 64, 64),
+            *args,
+            **kwargs,
+        )
+
+    # Used for checkpoints, tensorboard logging and env monitoring
+    experiment_path = f"{FLAGS.base_dir}/{FLAGS.mava_id}"
+
+    # Log every [log_every] seconds.
+    log_every = 10
+
+    # Pass custom logger config for evaluator to use.
+    # A custom json path can also be passed in.
+    logger_config = {
+        "evaluator": {
+            "to_json": True,
+        },
+    }
+
+    logger_factory = functools.partial(
+        logger_utils.make_logger,
+        directory=FLAGS.base_dir,
+        to_terminal=True,
+        to_tensorboard=True,
+        time_stamp=FLAGS.mava_id,
+        time_delta=log_every,
+    )
+
+    # Optimisers.
+    policy_optimiser = optax.chain(
+        optax.clip_by_global_norm(40.0), optax.scale_by_adam(), optax.scale(-1e-4)
+    )
+
+    critic_optimiser = optax.chain(
+        optax.clip_by_global_norm(40.0), optax.scale_by_adam(), optax.scale(-1e-4)
+    )
+
+    # Create the system.
+    system = ippo.IPPOSystem()
+
+    # Build the system.
+    system.build(
+        environment_factory=environment_factory,
+        network_factory=network_factory,
+        experiment_path=experiment_path,
+        policy_optimiser=policy_optimiser,
+        critic_optimiser=critic_optimiser,
+        run_evaluator=True,
+        sample_batch_size=5,
+        num_epochs=15,
+        num_executors=1,
+        multi_process=True,
+        normalise_observations=False,
+        normalise_target_values=False,
+        evaluation_interval={"executor_steps": 10000},
+        evaluation_duration={"evaluator_episodes": 32},
+        logger_factory=logger_factory,
+        logger_config=logger_config,
+        # Flag to activate the calculation of the absolute metric
+        absolute_metric=True,
+        # How many episodes the evaluator will run for
+        absolute_metric_duration=32,
+        # When the system will terminate and the absolute metric
+        # will be calculated
+        absolute_metric_interval=50000,
```

Comment:
Can we not use the `termination_condition` dictionary here?

Reply:
I will give it a try